cooperator/models/operation_request.py

```diff
@@ -135,12 +127,26 @@ def _compute_subscription_amount(self):
     def create(self, vals_list):
         records = super().create(vals_list)
         for record in records:
-            if (
-                record.operation_type == "transfer"
-                and record.subscription_request
-                and record.subscription_request.state != "transfer"
-            ):
-                record.subscription_request.state = "transfer"
+            if record.subscription_request:
+                if not record.subscription_request.is_operation:
+                    raise ValidationError(
+                        _(
+                            "The is_operation field of the subscription "
+                            "request must be true"
+                        )
+                    )
+                if record.subscription_request.source != "operation":
+                    raise ValidationError(
+                        _(
+                            "The source field of the subscription request "
+                            "must be set to operation"
+                        )
+                    )
```

Comment:
good point. i’ve found that the fields `is_operation`, `source` (and `state` for transfer operations) were related to operations, and instead of changing the way subscription requests work, i added this here to make sure the ones used by operations are uniform. i think it would be a good idea to rethink these fields at some point.